### PR TITLE
Add Cancun hard-fork settings for Chiado testnet

### PR DESCRIPTION
**Pull request by rubo**
## Changes

- Added Cancun transition timestamps to Chiado chain spec
- Updated tests
- Removed EIP-4788 `BEACON_ROOTS_ADDRESS` from chain specs

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [x] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

#### Requires documentation update

- [ ] Yes
- [x] No

#### Requires explanation in Release Notes

- [x] Yes
- [ ] No

 > For the Chiado network operators, this is a must-have release because of the upcoming Dencun hard-fork. We urge you to update promptly to ensure seamless network functionality post-fork.
>
> The Chiado Dencun hard-fork is scheduled at **6:15:40 PM UTC on Jan 31, 2024**.

### CODE REVIEW

File: src/Nethermind/Nethermind.Network.Test/ForkInfoTests.cs [hunk 1]
```diff
@@ -159,8 +159,9 @@ public void Fork_id_and_hash_as_expected_on_gnosis(long head, ulong headTimestam
     }
 
     [TestCase(0L, 0UL, "0x50d39d7b", ChiadoSpecProvider.ShanghaiTimestamp, "Chiado genesis")]
-    [TestCase(3945317, ChiadoSpecProvider.ShanghaiTimestamp, "0xa15a4252", 0ul, "First Shanghai timestamp")]
-    [TestCase(9945337, 9984930320ul, "0xa15a4252", 0ul, "Future Shanghai timestamp")]
+    [TestCase(3945317, ChiadoSpecProvider.ShanghaiTimestamp, "0xa15a4252", ChiadoSpecProvider.CancunTimestamp, "First Shanghai timestamp")]
+    [TestCase(4_000_000, ChiadoSpecProvider.CancunTimestamp, "0x5fbc16bc", 0ul, "First Cancun timestamp")]
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network.Test/ForkInfoTests.cs:163]: Can you confirm fork hash with Erigon?

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Network.Test/ForkInfoTests.cs (hunk 1 above)
Comment (rubo) [reply src/Nethermind/Nethermind.Network.Test/ForkInfoTests.cs:163]: Not now. I'm gonna do that after the timestamp is confirmed.

File: src/Nethermind/Nethermind.Specs.Test/ChainSpecStyle/ChainSpecBasedSpecProviderTests.cs [hunk 2]
```diff
@@ -224,7 +224,9 @@ public static IEnumerable<TestCaseData> ChiadoActivations
             yield return new TestCaseData((ForkActivation)(1, 20)) { TestName = "(1, 20)" };
             yield return new TestCaseData((ForkActivation)(1, ChiadoSpecProvider.ShanghaiTimestamp - 1)) { TestName = "Before Shanghai" };
             yield return new TestCaseData((ForkActivation)(1, ChiadoSpecProvider.ShanghaiTimestamp)) { TestName = "Shanghai" };
-            yield return new TestCaseData(new ForkActivation(1, GoerliSpecProvider.ShanghaiTimestamp + 100000000)) { TestName = "Future" };
+            yield return new TestCaseData((ForkActivation)(1, ChiadoSpecProvider.CancunTimestamp - 1)) { TestName = "Before Cancun" };
+            yield return new TestCaseData((ForkActivation)(1, ChiadoSpecProvider.CancunTimestamp)) { TestName = "Cancun" };
+            yield return new TestCaseData((ForkActivation)(1, GoerliSpecProvider.CancunTimestamp + 100000000)) { TestName = "Future" };
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Specs.Test/ChainSpecStyle/ChainSpecBasedSpecProviderTests.cs:229]: GoerliSpecProvider.CancunTimestamp a small bug

File: src/Nethermind/Chains/chiado.json [hunk 3]
```diff
@@ -69,6 +68,12 @@
     "eip3855TransitionTimestamp": "0x646e0e4c",
     "eip3860TransitionTimestamp": "0x646e0e4c",
     "eip4895TransitionTimestamp": "0x646e0e4c",
+    "eip1153TransitionTimestamp": "0x65ba8e4c",
+    "eip4788TransitionTimestamp": "0x65ba8e4c",
+    "eip4844TransitionTimestamp": "0x65ba8e4c",
+    "eip5656TransitionTimestamp": "0x65ba8e4c",
+    "eip6780TransitionTimestamp": "0x65ba8e4c",
+    "eip4788ContractAddress": "0x000F3df6D732807Ef1319fB7B8bB8522d0Beac02",
```
Comment (MarekM25) [inline src/Nethermind/Chains/chiado.json:76]: 4788 address doesn't have to be specified

Comment (rubo) [reply src/Nethermind/Chains/chiado.json:76]: Then why do we have it specified for Goerli?

File: src/Nethermind/Nethermind.Network.Test/ForkInfoTests.cs (hunk 1 above)
Comment (rubo) [reply src/Nethermind/Nethermind.Network.Test/ForkInfoTests.cs:163]: Confirmed

**Pull request merged**